include/fmt/format.h

```diff
@@ -1073,11 +1073,21 @@ inline auto equal2(const char* lhs, const char* rhs) -> bool {
 }
 
 // Copies two characters from src to dst.
-template <typename Char> void copy2(Char* dst, const char* src) {
-  *dst++ = static_cast<Char>(*src++);
-  *dst = static_cast<Char>(*src);
+template <typename Char>
+FMT_CONSTEXPR20 void copy2(Char* dst, const char* src) {
+  char dc0 = *src++;
+  char dc1 = *src;
```

Comment:
No need to introduce these variables.

Reply:
All the same, this gives the compiler the ability to generate more efficient code. (Although he may not use it)

Reply:
Could you give an example on godbolt illustrating this?

Reply:
https://godbolt.org/z/YKvoshP8K This is not enough? We untie the compiler's hands without losing anything.

Reply:
Interesting. Could you add a short comment clarifying this because otherwise it might be tempting to remove these seemingly redundant variables?

Reply:
logical.

Reply:
fix